### ISSUE TITLE
Accessing value of newly added EntityValue to an existing entity

### ISSUE THREAD

**Issue by anotherstarburst** [komlasapaty/datastore-entity#2]
Hey. This is a cool project and I've enjoyed playing with it. It makes interactions with datastore quite a bit easier (similar to ndb days of python2 in GAE)!

Question on table alters. If we start with a model like:

```
class User(DatastoreEntity):
    username = EntityValue(None)
```

and create an entity. And then we change the model to be 

```
class User(DatastoreEntity):
    username = EntityValue(None)
    email = EntityValue(None)
```

Then for an existing entity

```
> print(user.email)
<datastore_entity.entity_value.EntityValue object at 0x1019d2b80>
```

where I was expecting it to return `None`. Looking at the entity a bit more I can see that:

```
> print(user.email.__dict__)
{'value': None}

print(user.email.value)
None
```

So the value is there, it's just not being accessed correctly from what I can tell.  Is there a better way to do this or somewhere I can look to add a fix?

**Comment by komlasapaty** [comment]
Because entities of the same `kind` can have different properties(i.e different data types and even different property names), 'alter' will not work the same way as that of a relational database.

Datastore allows properties of an entity to be added 'willy-nilly'.

Remember that unlike an SQL record, to update an entity property, you basically have to read the whole entity, modify that single property and  then insert it back with all properties(including the unmodified ones). 

In your example, the entity you saved using a previous model definition of only a 'username' as a property will only have 'username' as the property. If you update(read alter) the model definition to add an 'email' property, the previously saved entity will not have an 'email' property at all. 

If however you want to add extra properties that are not in the model definition, the `save()` method has an `extra_props` keyword argument that takes a dictionary. When saving an entity, you can pass this value and that instance of the entity will be the ONLY one with these extra properties. This is a flexibility datastore provides.

There's also an `exclude` keyword argument that removes a property from an entity(even if property is in the model definition).

In other words, to remove/add properties to an entity record, you'll have to read those entities, modify them and save them back.

Having said this, there's typo in the documentation for the `save()` method where `extra_props` is referred to as `extra_data`. 
This will be fixed Thanks!

There's a chance I completely misunderstood you. Kindly let me know if this is the case.

**Comment by anotherstarburst** [comment]
Thanks for replying! 

You understand the issue perfectly. I realise that different entities of the same kind can have different properties. 

I guess what I'm suggesting (in a poorly phrased way) is that if a property doesn't exist for a given entity it is more useful to return `None` than to return an `EntityValue` object.

It'd be the difference between:

```
for user in users:
    if user.email:
        send_email()
```

and 

```
for user in users:
    if user.email and not isinstance(user.email, EntityValue):
        send_email()
```

**Comment by komlasapaty** [comment]
Yeah, on Datastore, `None` is itself a value, I thought returning it for a 'missing' property will not be accurate. It means one will not be able to differentiate between a missing property and a property with an actual `None` value.

**Comment by anotherstarburst** [comment]
That's a good point. I guess the decision needs to be made between accuracy and convenience... Personally I'd go for convenience, but as architect of this project it's your call :)
